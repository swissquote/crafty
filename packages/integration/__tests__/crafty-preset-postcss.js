/* global jest, describe, it, expect */

const configuration = require("@swissquote/crafty/src/configuration");
const getCommands = require("@swissquote/crafty/src/commands/index");

const testUtils = require("../utils");

// Add a high timeout because of https://github.com/facebook/jest/issues/8942
// Tests would be unreliable if they timeout >_<
jest.setTimeout(30000);

const getCrafty = configuration.getCrafty;

it("Loads crafty-preset-postcss and does not register gulp tasks", () => {
  const crafty = getCrafty(["@swissquote/crafty-preset-postcss"], {});

  const loadedPresets = crafty.config.loadedPresets.map(
    preset => preset.presetName
  );
  expect(loadedPresets).toContain("@swissquote/crafty-preset-postcss");

  const commands = getCommands(crafty);
  expect(Object.keys(commands)).toContain("cssLint");

  crafty.createTasks();
  expect(Object.keys(crafty.undertaker._registry.tasks())).toEqual([]);
});

it("Lints with the command", async () => {
  const cwd = await testUtils.getCleanFixtures(
    "crafty-preset-postcss/no-bundle"
  );

  const result = await testUtils.run(["cssLint", "css/*.scss"], cwd);

  expect(result).toMatchSnapshot();

  expect(testUtils.exists(cwd, "dist")).toBeFalsy();
});

it("Lints with the command in legacy mode", async () => {
  const cwd = await testUtils.getCleanFixtures(
    "crafty-preset-postcss/no-bundle"
  );

  const result = await testUtils.run(
    ["cssLint", "css/*.scss", "--preset", "legacy"],
    cwd
  );

  expect(result).toMatchSnapshot();

  expect(testUtils.exists(cwd, "dist")).toBeFalsy();
});

it("Lints with the command with custom config", async () => {
  const cwd = await testUtils.getCleanFixtures(
    "crafty-preset-postcss/no-bundle"
  );

  const result = await testUtils.run(
    ["cssLint", "css/*.scss", "--config", "stylelint.json"],
    cwd
  );

  expect(result).toMatchSnapshot();

  expect(testUtils.exists(cwd, "dist")).toBeFalsy();
});

it("Creates IDE Integration files", async () => {
  const cwd = await testUtils.getCleanFixtures("crafty-preset-postcss/ide", [
    "stylelint.config.js",
    "prettier.config.js",
    ".gitignore"
  ]);

  const result = await testUtils.run(["ide"], cwd);

  expect(result).toMatchSnapshot();
  expect(
    testUtils.readForSnapshot(cwd, "stylelint.config.js")
  ).toMatchSnapshot();

  expect(
    testUtils.readForSnapshot(cwd, "prettier.config.js")
  ).toMatchSnapshot();

  expect(testUtils.readForSnapshot(cwd, ".gitignore")).toMatchSnapshot();
});
