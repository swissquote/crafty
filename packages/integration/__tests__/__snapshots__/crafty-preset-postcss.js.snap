// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates IDE Integration files 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
Written prettier.config.js
Written stylelint.config.js
Written .gitignore
",
}
`;

exports[`Creates IDE Integration files 2`] = `
"// This configuration was generated by Crafty
// This file is generated to improve IDE Integration
// You don't need to commit this file, nor need it to run \`crafty build\`

module.exports = {
    \\"plugins\\": [
        \\"__PATH__/packages/stylelint-config-swissquote/index.js\\"
    ],
    \\"extends\\": [
        \\"__PATH__/packages/stylelint-config-swissquote/recommended.js\\"
    ],
    \\"rules\\": {
        \\"swissquote/no-type-outside-scope\\": null,
        \\"plugin/no-unsupported-browser-features\\": [
            true,
            {
                \\"severity\\": \\"warning\\",
                \\"browsers\\": \\"> 0.25%, Edge >= 17, Safari >= 11, iOS >= 11, Chrome >= 66, Firefox >= 60, IE >= 11, not op_mini all\\",
                \\"ignore\\": [
                    \\"rem\\",
                    \\"calc\\",
                    \\"css-initial-value\\",
                    \\"css-image-set\\",
                    \\"css-gradients\\"
                ]
            }
        ]
    }
};
"
`;

exports[`Creates IDE Integration files 3`] = `
"// This configuration was generated by Crafty
  // This file is generated to improve IDE Integration
  // You don't need to commit this file

  module.exports = {
    \\"overrides\\": [
        {
            \\"files\\": [
                \\"*.css\\",
                \\"*.scss\\"
            ],
            \\"options\\": {
                \\"tabWidth\\": 4
            }
        }
    ]
};
"
`;

exports[`Creates IDE Integration files 4`] = `
"/prettier.config.js
/stylelint.config.js
"
`;

exports[`Lints with the command 1`] = `
Object {
  "status": 2,
  "stdall": "
[__:__:__] Starting Crafty __version__...

css/imported.scss
 1:1  ✖  Types are allowed only inside a scope  swissquote/no-type-outside-scope

css/style.scss
  4:1   ✖  Types are allowed only inside a scope                                 swissquote/no-type-outside-scope
  6:14  ✖  Unexpected unit                                                       length-zero-no-unit
 10:1   ✖  Expected class selector \\".notAComponent\\" to match specified pattern   selector-class-pattern
 15:1   ✖  A state must be linked to a component                                 swissquote/no-state-without-component
 20:1   ✖  Expected \\"#no-id\\" to have no more than 0 ID selectors                 selector-max-id
 28:13  ✖  Expected nesting depth to be no more than 2                           max-nesting-depth
 38:16  ✖  Insert \\"⏎\\"                                                            prettier/prettier

",
}
`;

exports[`Lints with the command in legacy mode 1`] = `
Object {
  "status": 2,
  "stdall": "
[__:__:__] Starting Crafty __version__...

css/style.scss
  6:14  ✖  Unexpected unit   length-zero-no-unit
 38:16  ✖  Insert \\"⏎\\"        prettier/prettier

",
}
`;

exports[`Lints with the command with custom config 1`] = `
Object {
  "status": 2,
  "stdall": "
[__:__:__] Starting Crafty __version__...

css/style.scss
  6:14  ✖  Unexpected unit                                                       length-zero-no-unit
 10:1   ✖  Expected class selector \\".notAComponent\\" to match specified pattern   selector-class-pattern
 15:1   ✖  A state must be linked to a component                                 swissquote/no-state-without-component
 20:1   ✖  Expected \\"#no-id\\" to have no more than 0 ID selectors                 selector-max-id
 28:13  ✖  Expected nesting depth to be no more than 2                           max-nesting-depth
 38:16  ✖  Insert \\"⏎\\"                                                            prettier/prettier

",
}
`;
